src/main/java/week1/db/SqlUserServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+            changeAddressId(userId, newAddress);
+
+            return getUserById(userId);
+        }
+    }
+
+    @Override
+    public void delete(Integer userId) throws SQLException {
+
+        preparedStatement = connection.prepareStatement
+                ("DELETE FROM users WHERE id = ?");
+
+        preparedStatement.setInt(1, userId);
+        preparedStatement.execute();
+
+    }
+
+    private void changeAddressId(Integer userId, Address newAddress) throws SQLException {
+
+        String selectSQL = "UPDATE users SET address_id = ? WHERE id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(selectSQL);
+        preparedStatement.setInt(1, newAddress.getId());
+        preparedStatement.setInt(2, userId);
+    }
+
+    private User getUserById(Integer userId) throws SQLException {
+
+        User user = new User();
+
+        String selectSQL = "SELECT * FROM users WHERE id =  ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(selectSQL);
+        preparedStatement.setInt(1, userId);
+        ResultSet resultSet = preparedStatement.executeQuery(selectSQL);
+
+        while (resultSet.next()) {
+
+            user.setId(resultSet.getInt("id"));
+            user.setFirst_name(resultSet.getString("first_name"));
+            user.setLast_name(resultSet.getString("last_name"));
+            user.setAge(resultSet.getInt("age"));
+            user.setAddress(getAddressById(resultSet.getInt("address_id")));
+        }
+
+        return user;
+    }
+
+    private List<User> createUserListFromSet(ResultSet resultSet) throws SQLException {
+
+        List<User> userList = new ArrayList<>();
+
+        while (resultSet.next()) {
+
+            userList.add(new User(
+                            resultSet.getInt("id"),
+                            resultSet.getString("first_name"),
+                            resultSet.getString("last_name"),
+                            resultSet.getInt("age"),
+                            new Address(getAddressById(resultSet.getInt("address_id")))
+                    )
+            );
+        }
+        return userList;
+    }
+
+    private Address getAddressById(int address_id) throws SQLException {
+
+        Address address = new Address();
+
+        String selectSQL = "SELECT * FROM address WHERE address_id =  ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(selectSQL);
+        preparedStatement.setInt(1, address_id);
+        ResultSet resultSet = preparedStatement.executeQuery(selectSQL);
+
+        while (resultSet.next()) {
+
+            address.setId(resultSet.getInt("id"));
+            address.setCity(resultSet.getString("city"));
+            address.setStreet(resultSet.getString("street"));
+            address.setHouse(resultSet.getInt("house"));
+        }
+
+        return address;
+    }
+
+    private int setAddressId(String city, String street, int house) throws SQLException {
```

Comment:
I would change the method name as in fact you are not setting address id here, you are evaluating it to pass as a parameter to another method.

Reply:
can u pls advise a name for this method

I thought that it is a kind of setter for the id)

Reply:
thanks